addons/mail/models/update.py

```diff
@@ -98,7 +98,7 @@ def update_notification(self, cron_mode=True):
             # old behavior based on res.log; now on mail.message, that is not necessarily installed
             user = self.env['res.users'].sudo().browse(SUPERUSER_ID)
             poster = self.sudo().env.ref('mail.channel_all_employees')
-            if not (poster and poster.exists()):
+            if not poster:
                 if not user.exists():
                     return True
                 poster = user
```

Comment:
```suggestion
            poster = self.sudo().env.ref('mail.channel_all_employees')
```
probably, really: the `ref` was originally a `xml_to_object` but the raise flag was  not flipped so for the last 6 years it's raised an error if someone had deleted the reference object, and apparently nobody ever cared.

Not to mention handling the case where somebody deleted the superuser which... wow.

Might be worth waiting for @tde-banana-odoo to confirm that's fine since it's in mail, but...

Reply:
There _exists_ code to avoid deleting the superuser.  This should not happen, therefore that error should not be handled.  :+1: on @xmo-odoo 's suggestion.

Reply:
Yeah, it is a completely useless condition. If there are no poster, it will raise during the `ref` call anyway.